### ISSUE TITLE
ERROR:utils:Step must be of type int or List[int]

### ISSUE THREAD

**Issue by HauHe** [KTH-dESA/OSeMOSYS_step#42]
When running `main.py` with `python main.py --step_length 5` I get an error saying that the there is no sol file produced from step 0. In the log file I see the below message:
`ERROR:utils:Step must be of type int or List[int]. Recieved type <class 'tuple'>
WARNING:otoole.read_strategies:Parameter ResultsPath could not be found in the configuration.`

**Comment by HauHe** [comment]
When calling with `python main.py --step_length 1 --step_length 5`
I get:
`Building and Solving Models:   0%|          | 0/5 [00:00<?, ?it/s]                                                             
Traceback (most recent call last):
  File "/Users/haukeh/GitHub/HauHe/OSeMOSYS_step/src/main.py", line 538, in <module>
    main() #input_data,step_length,path_param,solver)
    ^^^^^^
  File "/Users/haukeh/opt/miniconda3/envs/osemosys_step-tervor-update/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/haukeh/opt/miniconda3/envs/osemosys_step-tervor-update/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/haukeh/opt/miniconda3/envs/osemosys_step-tervor-update/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/haukeh/opt/miniconda3/envs/osemosys_step-tervor-update/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/haukeh/GitHub/HauHe/OSeMOSYS_step/src/main.py", line 194, in main
    preprocess_data.main("otoole", str(data_file), str(data_file_pp))
  File "/Users/haukeh/GitHub/HauHe/OSeMOSYS_step/src/preprocess_data.py", line 134, in main
    start_year = year_list[0]
                 ~~~~~~~~~^^^
IndexError: list index out of range`

But log file content remains the same

**Comment by trevorb1** [comment]
Hi @HauHe! The  `ERROR:utils:Step must be of type int or List[int]. Recieved type <class 'tuple'>` was a legacy error message that has been switched to an info level. The code should still run past this point though. 

Whats the exact error message for the no `.sol` file? 

For the second error message, it seems like the years may not be getting parsed correctly. Maybe print out what the years are being returned from the `split_data()` function?

**Comment by trevorb1** [comment]
Also, for the no `.sol` error. Make sure you double check the snakefile is being called correctly (ie. try `snakemake -c`) and no error is being raised!

**Comment by HauHe** [comment]
The error message for the no .sol file error is:
`Building and Solving Models:   0%|          | 0/5 [00:00<?, ?it/s]                                                             
Traceback (most recent call last):
  File "/Users/haukeh/GitHub/HauHe/OSeMOSYS_step/src/main.py", line 538, in <module>
    main() #input_data,step_length,path_param,solver)
    ^^^^^^
  File "/Users/haukeh/opt/miniconda3/envs/osemosys_step-tervor-update/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/haukeh/opt/miniconda3/envs/osemosys_step-tervor-update/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/haukeh/opt/miniconda3/envs/osemosys_step-tervor-update/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/haukeh/opt/miniconda3/envs/osemosys_step-tervor-update/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/haukeh/GitHub/HauHe/OSeMOSYS_step/src/main.py", line 323, in main
    if solve.check_cbc_feasibility(str(sol_file)) == 1:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/haukeh/GitHub/HauHe/OSeMOSYS_step/src/solve.py", line 91, in check_cbc_feasibility
    with open(sol) as f:
         ^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '../steps/step_0/model.sol'`

**Comment by HauHe** [comment]
Ok, mistake the error when only providing one `step_length` doesn't occur when having snakemake installed...

But the error when providing two `--step_length` persists.

**Comment by HauHe** [comment]
I will have a look at it, but also only next week. Have a deadline on Monday.

**Comment by HauHe** [comment]
Hi @trevorb1,
I think I fixed the error. I merged my fix into the devlop branch. 
If you have a few minutes it would be great if you could have a look.
It tells me now that some of the models didn't solve. But I guess that was something I put into the data.

**Comment by trevorb1** [comment]
Hey @HauHe! I fixed the snakemake issue we discovered earlier today! Also, I sorted out the solving logs, so thats been added back in! (See this branch https://github.com/KTH-dESA/OSeMOSYS_step/pull/43) 

We are still getting a couple failed solves, so I think there are some issues to sort out. Did you want to rerun this and check the production between the old and new models again using your plotting script?
